chrimcha/ThirdPlace
# 9 admin dashboard

### Summary:
- Added `role` field to `User` model with getters and setters
- Refactored code in `AuthenticationController` to register new `Users` with a default `role` of "Basic"
- Created `AdminDashboard` and added route path in `main.jsx`
- Created `UsersTable` component to display all users in DB
- Added link to `AdminDashboard`  that only appears if `User` `role` is "Admin"
- Replace "Password" with "Role" in `ProfileInfoCard`
- Fixed typo and added comments in `UserServices`

### To Test:
- [x] Login User on site
- [x] In the MySQL Workbench, change your users role to "Admin" with your user id:
```
update users
set role = "Admin"
where id = 4;
```
- [x] Go back to site and refresh the page, you should now see a gear emoji in the navbar
- [x] Click on that gear to get to Admin Dashboard (you will get a `Unexpected Application Error!
Cannot read properties of undefined (reading 'map')`, reload the page to get to see dashboard)

> The dashboard currently contains a table of all users in BD and titles for the next two sections
>
>Buttons are not hooked up yet and I will be fixing the load error in the next PR
